great_expectations/execution_engine/sqlalchemy_dialect.py

```diff
@@ -27,6 +27,9 @@ class GESqlDialect(enum.Enum):
     def __eq__(self, other: Union[str, GESqlDialect]):
```

Comment:
Removing dunder methods and adding `str` mixin doesn't work because we want to allow for equivalence with byte strings as well (e.g. b"hive").